stan/math/prim/mat/fun/divide_columns.hpp

```diff
@@ -49,17 +24,19 @@ inline typename std::vector<Eigen::Matrix<T_x, R, C>> divide_columns(
  *
  */
 template <typename T_x, typename T_v, int R, int C>
-inline typename std::vector<Eigen::Matrix<T_x, R, C>> divide_columns(
-    const std::vector<Eigen::Matrix<T_x, R, C>> &x,
-    const std::vector<T_v> &vec) {
+inline typename std::vector<
+    Eigen::Matrix<typename return_type<T_x, T_v, double>::type, R, C>>
+divide_columns(const std::vector<Eigen::Matrix<T_x, R, C>> &x,
+               const std::vector<T_v> &vec) {
   size_t N = x.size();
   size_t D = x[0].size();
   check_size_match("divide_columns", "x dimension", D, "vector", vec.size());
```

Comment:
Can check_size_match move inside the for loop over n? That way the size of each column vector is checked. It may seem annoying but I think we usually try to perform all the size checks we can (see: https://github.com/stan-dev/math/blob/develop/stan/math/prim/mat/prob/multi_normal_lpdf.hpp#L51 )

Reply:
good catch this is more robust